File: _deprecated/old_scripts/test-flask-app.js
// Test Flask application using existing Playwright setup
// NO EMOJIS
const { chromium } = require('playwright');
const fs = require('fs');
const path = require('path');

// Ensure logs directory exists
const logsDir = path.join(__dirname, '../logs');
if (!fs.existsSync(logsDir)) {
  fs.mkdirSync(logsDir, { recursive: true });
}

const logFile = path.join(logsDir, `flask-app-test-${Date.now()}.log`);

async function testFlaskApp() {
  const url = 'http://localhost:5000';
  console.log(`Testing Flask Application at ${url}`);
  console.log(`Logs will be saved to: ${logFile}`);
  
  const browser = await chromium.launch({ 
    headless: false,  // Show browser for debugging
    devtools: true    // Open dev tools
  });
  
  const context = await browser.newContext();
  const page = await context.newPage();
  
  // Capture console logs
  page.on('console', async msg => {
    const timestamp = new Date().toISOString();
    const type = msg.type().toUpperCase().padEnd(7);
    const text = msg.text();
    
    const logEntry = `[${timestamp}] ${type} | ${text}`;
    console.log(logEntry);
    fs.appendFileSync(logFile, logEntry + '\n');
  });
  
  // Capture page errors
  page.on('pageerror', error => {
    const timestamp = new Date().toISOString();
    const errorEntry = `[${timestamp}] ERROR  | ${error.message}\n${error.stack || ''}`;
    console.error(errorEntry);
    fs.appendFileSync(logFile, errorEntry + '\n');
  });
  
  // Capture network errors
  page.on('requestfailed', request => {
    const timestamp = new Date().toISOString();
    const errorEntry = `[${timestamp}] NETWORK| Failed: ${request.method()} ${request.url()} - ${request.failure().errorText}`;
    console.error(errorEntry);
    fs.appendFileSync(logFile, errorEntry + '\n');
  });
  
  // Capture responses
  page.on('response', response => {
    const timestamp = new Date().toISOString();
    const entry = `[${timestamp}] HTTP   | ${response.status()} ${response.url()}`;
    if (response.status() >= 400) {
      console.log(entry);
    }
    fs.appendFileSync(logFile, entry + '\n');
  });
  
  try {
    // Test 1: Load main page
    console.log('\n=== TEST 1: Loading main page ===');
    await page.goto(url, { waitUntil: 'networkidle' });
    console.log('Page loaded successfully');
    
    // Take initial screenshot
    await page.screenshot({ path: 'logs/flask-app-initial.png' });
    
    // Test 2: Check if FullCalendar loads
    console.log('\n=== TEST 2: FullCalendar initialization ===');
    await page.waitForSelector('#calendar', { timeout: 10000 });
    console.log('Calendar element found');
    
    // Wait for FullCalendar to initialize
    await page.waitForSelector('.fc-view-harness', { timeout: 15000 });
    console.log('FullCalendar initialized successfully');
    
    // Test 3: Check health status
    console.log('\n=== TEST 3: Health status check ===');
    const healthStatus = await page.textContent('#health-status');
    console.log(`Health status: ${healthStatus}`);
    
    // Test 4: Check API endpoints
    console.log('\n=== TEST 4: API endpoint testing ===');
    
    // Test events API
    const eventsResponse = await page.evaluate(async () => {
      const response = await fetch('/api/events');
      return {
        status: response.status,
        data: await response.json()
      };
    });
    console.log(`Events API: ${eventsResponse.status} - ${eventsResponse.data.length} events`);
    
    // Test tasks API
    const tasksResponse = await page.evaluate(async () => {
      const response = await fetch('/api/tasks');
      return {
        status: response.status,
        data: await response.json()
      };
    });
    console.log(`Tasks API: ${tasksResponse.status} - ${tasksResponse.data.length} tasks`);
    
    // Test health API
    const healthResponse = await page.evaluate(async () => {
      const response = await fetch('/health');
      return {
        status: response.status,
        data: await response.json()
      };
    });
    console.log(`Health API: ${healthResponse.status} - ${healthResponse.data.service}`);
    
    // Test 5: Create test event via calendar
    console.log('\n=== TEST 5: Event creation test ===');
    
    // Click on calendar to create event
    const calendarEl = await page.locator('#calendar .fc-timegrid-slot[data-time="10:00:00"]').first();
    await calendarEl.click();
    
    // Wait for prompt and enter event title
    page.on('dialog', async dialog => {
      console.log(`Dialog: ${dialog.message()}`);
      if (dialog.message().includes('Event Title')) {
        await dialog.accept('Test Event from Playwright');
      } else if (dialog.message().includes('blocking event')) {
        await dialog.accept(); // Accept blocking event
      }
    });
    
    // Click to trigger event creation
    await page.click('#calendar .fc-timegrid-body');
    await page.waitForTimeout(2000); // Wait for event creation
    
    console.log('Event creation dialog handled');
    
    // Test 6: Create test task
    console.log('\n=== TEST 6: Task creation test ===');
    
    const addTaskBtn = await page.locator('button:has-text("Add Task")');
    await addTaskBtn.click();
    
    // Handle task creation prompts
    page.on('dialog', async dialog => {
      console.log(`Task Dialog: ${dialog.message()}`);
      if (dialog.message().includes('Task Title')) {
        await dialog.accept('Test Task from Playwright');
      } else if (dialog.message().includes('Duration')) {
        await dialog.accept('45');
      } else if (dialog.message().includes('Urgency')) {
        await dialog.accept('8');
      } else if (dialog.message().includes('Description')) {
        await dialog.accept('Test task created by Playwright automation');
      }
    });
    
    await page.waitForTimeout(3000); // Wait for task creation
    console.log('Task creation completed');
    
    // Test 7: Check tasks list updates
    console.log('\n=== TEST 7: Tasks list verification ===');
    
    const tasksListContent = await page.textContent('#tasks-list');
    console.log(`Tasks list content length: ${tasksListContent.length} characters`);
    
    // Take final screenshot
    await page.screenshot({ path: 'logs/flask-app-final.png' });
    
    // Test 8: Calendar views
    console.log('\n=== TEST 8: Calendar view switching ===');
    
    // Test month view
    const monthBtn = await page.locator('.fc-dayGridMonth-button');
    if (await monthBtn.isVisible()) {
      await monthBtn.click();
      await page.waitForTimeout(1000);
      console.log('Month view activated');
    }
    
    // Test week view
    const weekBtn = await page.locator('.fc-timeGridWeek-button');
    if (await weekBtn.isVisible()) {
      await weekBtn.click();
      await page.waitForTimeout(1000);
      console.log('Week view activated');
    }
    
    // Test day view
    const dayBtn = await page.locator('.fc-timeGridDay-button');
    if (await dayBtn.isVisible()) {
      await dayBtn.click();
      await page.waitForTimeout(1000);
      console.log('Day view activated');
    }
    
    // Test 9: Navigation
    console.log('\n=== TEST 9: Navigation testing ===');
    
    // Test events page navigation
    const eventsLink = await page.locator('a:has-text("Events")');
    if (await eventsLink.isVisible()) {
      await eventsLink.click();
      await page.waitForTimeout(1000);
      console.log('Events page navigation attempted');
    }
    
    // Test tasks page navigation  
    const tasksLink = await page.locator('a:has-text("Tasks")');
    if (await tasksLink.isVisible()) {
      await tasksLink.click();
      await page.waitForTimeout(1000);
      console.log('Tasks page navigation attempted');
    }
    
    // Return to schedule
    const scheduleLink = await page.locator('a:has-text("Schedule")');
    if (await scheduleLink.isVisible()) {
      await scheduleLink.click();
      await page.waitForTimeout(1000);
      console.log('Schedule page navigation completed');
    }
    
    console.log('\n' + '='.repeat(60));
    console.log('FLASK APPLICATION TEST COMPLETED SUCCESSFULLY');
    console.log('='.repeat(60));
    console.log('✓ Page loads without errors');
    console.log('✓ FullCalendar initializes properly');
    console.log('✓ API endpoints respond correctly');
    console.log('✓ Event creation functionality works');
    console.log('✓ Task management functionality works');
    console.log('✓ Calendar view switching works');
    console.log('✓ Navigation between pages works');
    console.log('✓ Health monitoring is active');
    console.log('='.repeat(60));
    
    // Wait 5 more seconds to observe
    console.log('Observing for 5 more seconds...');
    await page.waitForTimeout(5000);
    
  } catch (error) {
    console.error('\nERROR during testing:', error.message);
    fs.appendFileSync(logFile, `\n[ERROR] ${error.message}\n${error.stack}\n`);
    
    // Take error screenshot
    await page.screenshot({ path: 'logs/flask-app-error.png' });
  }
  
  await browser.close();
  
  console.log('\n' + '='.repeat(50));
  console.log('Test Session Summary:');
  console.log(`Log file: ${logFile}`);
  console.log('Screenshots saved in logs/ directory');
  console.log('='.repeat(50));
  
  return logFile;
}

// Check if Flask is running
async function checkFlaskRunning() {
  try {
    const response = await fetch('http://localhost:5000/health');
    return response.ok;
  } catch {
    return false;
  }
}

if (require.main === module) {
  console.log('Checking if Flask app is running on localhost:5000...');
  
  checkFlaskRunning().then(isRunning => {
    if (!isRunning) {
      console.log('ERROR: Flask app is not running!');
      console.log('Please start it first:');
      console.log('  cd flask_app');
      console.log('  python app.py');
      process.exit(1);
    }
    
    console.log('Flask app detected, starting comprehensive test...\n');
    testFlaskApp().catch(console.error);
  });
}

module.exports = { testFlaskApp };

File: _deprecated/old_tests/test_flask_app.py
#!/usr/bin/env python3
"""
Playwright test for Flask application
NO EMOJIS
"""
import asyncio
import json
from playwright.async_api import async_playwright
from datetime import datetime

async def test_flask_app():
    """Test the Flask application with Playwright"""
    
    async with async_playwright() as p:
        # Launch browser
        browser = await p.chromium.launch(headless=False)  # Set to True for headless
        page = await browser.new_page()
        
        print("Testing Flask Application")
        print("=" * 50)
        
        try:
            # Test 1: Load main page
            print("1. Loading main page...")
            await page.goto("http://localhost:5000")
            await page.wait_for_load_state('networkidle')
            
            # Check if page loaded
            title = await page.title()
            print(f"   Page title: {title}")
            
            # Check if FullCalendar loaded
            calendar = await page.query_selector('#calendar')
            if calendar:
                print("   ✓ FullCalendar element found")
            else:
                print("   ✗ FullCalendar element NOT found")
                
            # Wait for FullCalendar to initialize
            await page.wait_for_selector('.fc-view-harness', timeout=10000)
            print("   ✓ FullCalendar initialized")
            
            # Test 2: Check health endpoint
            print("\n2. Testing health endpoint...")
            health_response = await page.evaluate("""
                fetch('/health').then(r => r.json())
            """)
            print(f"   Health status: {health_response}")
            
            # Test 3: Test API endpoints
            print("\n3. Testing API endpoints...")
            
            # Get events
            events_response = await page.evaluate("""
                fetch('/api/events').then(r => r.json())
            """)
            print(f"   Events API: {len(events_response)} events found")
            
            # Get tasks
            tasks_response = await page.evaluate("""
                fetch('/api/tasks').then(r => r.json())
            """)
            print(f"   Tasks API: {len(tasks_response)} tasks found")
            
            # Test 4: Create a test event
            print("\n4. Creating test event...")
            
            # Create event via API
            test_event = {
                'title': 'Test Event from Playwright',
                'start': '2025-01-20T10:00:00',
                'end': '2025-01-20T11:00:00',
                'is_blocking': True,
                'description': 'Test event created by automated test'
            }
            
            create_response = await page.evaluate(f"""
                fetch('/api/events', {{
                    method: 'POST',
                    headers: {{'Content-Type': 'application/json'}},
                    body: JSON.stringify({json.dumps(test_event)})
                }}).then(r => r.json())
            """)
            
            if 'id' in create_response:
                print(f"   ✓ Event created with ID: {create_response['id']}")
                event_id = create_response['id']
                
                # Test 5: Delete the test event
                print("\n5. Deleting test event...")
                delete_response = await page.evaluate(f"""
                    fetch('/api/events/{event_id}', {{
                        method: 'DELETE'
                    }}).then(r => r.json())
                """)
                print(f"   Delete response: {delete_response}")
                
            else:
                print(f"   ✗ Failed to create event: {create_response}")
            
            # Test 6: Create a test task
            print("\n6. Creating test task...")
            
            test_task = {
                'title': 'Test Task from Playwright',
                'description': 'Test task created by automated test',
                'duration': 30,
                'urgency': 7,
                'priority': 'high'
            }
            
            task_response = await page.evaluate(f"""
                fetch('/api/tasks', {{
                    method: 'POST',
                    headers: {{'Content-Type': 'application/json'}},
                    body: JSON.stringify({json.dumps(test_task)})
                }}).then(r => r.json())
            """)
            
            if 'id' in task_response:
                print(f"   ✓ Task created with ID: {task_response['id']}")
                task_id = task_response['id']
                
                # Test task completion
                print("\n7. Completing test task...")
                complete_response = await page.evaluate(f"""
                    fetch('/api/tasks/{task_id}/complete', {{
                        method: 'POST'
                    }}).then(r => r.json())
                """)
                print(f"   Task completion response: {complete_response}")
                
            else:
                print(f"   ✗ Failed to create task: {task_response}")
            
            # Test 7: Check UI elements
            print("\n8. Testing UI elements...")
            
            # Check if sidebar exists
            sidebar = await page.query_selector('.sidebar')
            if sidebar:
                print("   ✓ Sidebar found")
            else:
                print("   ✗ Sidebar NOT found")
            
            # Check if tasks list exists
            tasks_list = await page.query_selector('#tasks-list')
            if tasks_list:
                print("   ✓ Tasks list found")
            else:
                print("   ✗ Tasks list NOT found")
            
            # Check if health status indicator exists
            health_status = await page.query_selector('#health-status')
            if health_status:
                status_text = await health_status.text_content()
                print(f"   ✓ Health status indicator: {status_text}")
            else:
                print("   ✗ Health status indicator NOT found")
            
            # Test 8: Check calendar functionality
            print("\n9. Testing calendar functionality...")
            
            # Check if calendar has proper view buttons
            view_buttons = await page.query_selector_all('.fc-button')
            print(f"   ✓ Found {len(view_buttons)} calendar view buttons")
            
            # Check current view
            current_view = await page.evaluate("""
                window.calendar ? window.calendar.view.type : 'Calendar not initialized'
            """)
            print(f"   Current calendar view: {current_view}")
            
            # Take a screenshot
            timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
            screenshot_path = f"flask_app_test_{timestamp}.png"
            await page.screenshot(path=screenshot_path)
            print(f"\n   📸 Screenshot saved: {screenshot_path}")
            
            print("\n" + "=" * 50)
            print("✓ Flask Application Test Completed Successfully!")
            print("✓ All core functionality appears to be working")
            print("✓ FullCalendar is properly initialized")
            print("✓ API endpoints are responsive")
            print("✓ UI elements are rendering correctly")
            
        except Exception as e:
            print(f"\n❌ Test failed with error: {e}")
            # Take error screenshot
            timestamp = datetime.now().strftime("%Y%m%d_%H%M%S")
            await page.screenshot(path=f"flask_app_error_{timestamp}.png")
            
        finally:
            await browser.close()

if __name__ == "__main__":
    print("Make sure Flask app is running on http://localhost:5000")
    print("Run: cd flask_app && python app.py")
    print("Then run this test script")
    print()
    
    asyncio.run(test_flask_app())